great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -448,6 +448,10 @@ def concurrency(self) -> Optional[ConcurrencyConfig]:
     def assistants(self) -> DataAssistantDispatcher:
         return self._assistants
 
+    def set_config(self, project_config: DataContextConfig) -> None:
+        self._project_config = project_config
```

Comment:
A quick question. Why do we need 2 handles to `project_config` and will one we going away?

Reply:
it'll eventually go away! just keeping it there for legacy purposes

Reply:
Should we add a deprecation warning here?